### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ostUp**
[DEMO LINK](https://ostUp.github.io/react_tabs/)

### CODE REVIEW

**Review by Andrii-Dunets — APPROVED** [review]
Well done)

File: src/components/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -1 +1,52 @@
...
+  selectedTabId: string;
+  onTabSelected: (tab: Tab) => void;
+};
+
+const getTabById = (tabs: Tab[], tabId: string) => (
+  tabs.find(tab => tab.id === tabId) || tabs[0]
+);
+
+export const Tabs: React.FC<Props> = ({ tabs, selectedTabId, onTabSelected }) => {
+  const { id: validSelectedTabId, content } = getTabById(tabs, selectedTabId);
```
Comment (AlinaMatuschak) [inline src/components/Tabs/Tabs.tsx:18]:
```suggestion
export const Tabs: React.FC<Props> = ({
  tabs,
  selectedTabId,
  onTabSelected,
}) => {
  const {
    id: validSelectedTabId,
    content,
  } = getTabById(tabs, selectedTabId);
```

File: src/components/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -1 +1,52 @@
...
+
+  const handleClick = (id: string) => {
+    if (id !== validSelectedTabId) {
+      onTabSelected((getTabById(tabs, id)));
+    }
+  };
```
Comment (AlinaMatuschak) [inline src/components/Tabs/Tabs.tsx:24]:
```suggestion
  const handleClick = (tab: Tab) => {
    if (tab.id !== validSelectedTabId) {
      onTabSelected(tab);
    }
  };
```

File: src/components/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -1 +1,52 @@
...
+      <div className="tabs is-boxed">
+        <ul>
+          {tabs.map(({ id, title }) => (
+            <li
+              key={id}
+              className={classNames(
+                { 'is-active': id === validSelectedTabId },
+              )}
+              data-cy="Tab"
+              onClick={() => handleClick(id)}
```
Comment (AlinaMatuschak) [inline src/components/Tabs/Tabs.tsx:37]: you can here throw tab instead of tab id
```suggestion
              onClick={() => handleClick(id)}
```

**Review by AlinaMatuschak — APPROVED** [review]

**Pull request closed without merging**